core/templates/pages/learner-dashboard-page/learner-dashboard-page.component.ts

```diff
@@ -14,517 +14,611 @@
+        this.numberNonexistentCompletedExplorations = (
+          responseData.numberOfNonexistentActivities.completedExplorations);
+        this.numberNonexistentCompletedCollections = (
+          responseData.numberOfNonexistentActivities.completedCollections);
+        this.numberNonexistentExplorationsFromPlaylist = (
+          responseData.numberOfNonexistentActivities.explorationPlaylist);
+        this.numberNonexistentCollectionsFromPlaylist = (
+          responseData.numberOfNonexistentActivities.collectionPlaylist);
+        this.completedToIncompleteCollections = (
+          responseData.completedToIncompleteCollections);
+        this.threadSummaries = responseData.threadSummaries;
+        this.numberOfUnreadThreads =
+          responseData.numberOfUnreadThreads;
+        this.explorationPlaylist = responseData.explorationPlaylist;
+        this.collectionPlaylist = responseData.collectionPlaylist;
+        this.activeSection = (
+          LearnerDashboardPageConstants
+            .LEARNER_DASHBOARD_SECTION_I18N_IDS.INCOMPLETE);
+        this.activeSubsection = (
+          LearnerDashboardPageConstants
+            .LEARNER_DASHBOARD_SUBSECTION_I18N_IDS.EXPLORATIONS);
+        this.feedbackThreadActive = false;
+
+        this.noExplorationActivity = (
+          (this.completedExplorationsList.length === 0) &&
+            (this.incompleteExplorationsList.length === 0));
+        this.noCollectionActivity = (
+          (this.completedCollectionsList.length === 0) &&
+            (this.incompleteCollectionsList.length === 0));
+        this.noActivity = (
+          (this.noExplorationActivity) && (this.noCollectionActivity) &&
+          (this.explorationPlaylist.length === 0) &&
+          (this.collectionPlaylist.length === 0));
+      }, errorResponseStatus => {
+        if (
+          AppConstants.FATAL_ERROR_CODES.indexOf(errorResponseStatus) !== -1) {
+          this.alertsService.addWarning(
+            'Failed to get learner dashboard data');
         }
-      };
-
-      ctrl.setSubscriptionSortingOptions = function(sortType) {
-        if (sortType === ctrl.currentSubscribersSortType) {
-          ctrl.isCurrentSubscriptionSortDescending = (
-            !ctrl.isCurrentSubscriptionSortDescending);
-        } else {
-          ctrl.currentSubscribersSortType = sortType;
+      }
+    );
+
+
+    Promise.all([
+      userInfoPromise,
+      dashboardDataPromise,
+    ]).then(() => {
+      // The timeout is required because at execution time,
+      // the element may not be present in the DOM yet.Thus it ensure
+      // that the element is visible before focussing.
+      setTimeout(() => {
+        this.loaderService.hideLoadingScreen();
+        this.addFocusWithoutScroll('ourLessonsBtn');
+      }, 0);
+    // eslint-disable-next-line dot-notation
+    }).catch(errorResponse => {
+      // This is placed here in order to satisfy Unit tests.
```

Comment:
I don't understand this, please explain

Reply:
In the above Promise.all(...), two promises are used `userInfoPromise`, `dashboardDataPromise`.
If there is no error, then the loading screen will disappear and the learner's dashboard page will appear simultaneously.
In case of errors in any of those promises, without catch block it throws this error.
(I've reproduced the failure scenario for explanation)
![Screenshot from 2021-04-15 15-10-32](https://user-images.githubusercontent.com/54100702/114849532-78913100-9dfd-11eb-87f4-2b7c9a902c81.png)
And we don't need to handle anything is catch block because it has already been handled in the errorResponse of the promise.